test/e2e/storage/pd.go

```diff
@@ -325,30 +337,35 @@ var _ = SIGDescribe("Pod Disks", func() {
 				diskName, err := framework.CreatePDWithRetry()
 				framework.ExpectNoError(err, "Error creating a pd")
 
-				targetNode := &nodes.Items[0]
+				targetNode := &nodes.Items[0] // for node delete ops
 				host0Pod := testPDPod([]string{diskName}, host0Name, false, 1)
 				containerName := "mycontainer"
 
 				defer func() {
 					By("defer: cleaning up PD-RW test env")
 					framework.Logf("defer cleanup errors can usually be ignored")
-					if nodeOp == deleteNode {
-						podClient.Delete(host0Pod.Name, metav1.NewDeleteOptions(0))
-					}
+					By("defer: delete host0Pod")
+					podClient.Delete(host0Pod.Name, metav1.NewDeleteOptions(0))
+					By("defer: detach and delete PDs")
 					detachAndDeletePDs(diskName, []types.NodeName{host0Name})
-					if nodeOp == deleteNodeObj {
-						targetNode.ObjectMeta.SetResourceVersion("0")
-						// need to set the resource version or else the Create() fails
-						_, err := nodeClient.Create(targetNode)
-						framework.ExpectNoError(err, "defer: Unable to re-create the deleted node")
+					if disruptOp == deleteNode || disruptOp == deleteNodeObj {
+						if disruptOp == deleteNodeObj {
+							targetNode.ObjectMeta.SetResourceVersion("0")
+							// need to set the resource version or else the Create() fails
+							By("defer: re-create host0 node object")
+							_, err := nodeClient.Create(targetNode)
+							framework.ExpectNoError(err, fmt.Sprintf("defer: Unable to re-create the deleted node object %q", targetNode.Name))
```

Comment:
Just a note, since we talked about this earlier.  Using `framework.ExpectNoError` inside a defer is safe.  Typically, if 2 `Expect()` calls resolve false, only the 1st one is reported in the log.  The `framework.ExpectNoError()` wrapper calls `fmt.Logf()` if `err != nil` though, so it's not an issue.